packages/grid/_modules_/grid/hooks/utils/useResizeContainer.ts

```diff
@@ -8,6 +9,15 @@ export function useResizeContainer(apiRef): (size: ElementSize) => void {
   const gridLogger = useLogger('useResizeContainer');
   const { autoHeight } = useGridSelector(apiRef, optionsSelector);
 
+  const debounceResize = React.useMemo(
+    () =>
+      debounce((size: ElementSize) => {
```

Comment:
We can't use the debounce API exactly like this. It schedules a timeout that can fire after the component is unmounted. We should call clear(). For instance: https://github.com/mui-org/material-ui/blob/33850e52ca32787fbc02b64d3225bac2d451612c/packages/material-ui/src/Tabs/ScrollbarSize.js#L41

or even better. Here is the line of code you had removed: https://github.com/mui-org/material-ui-x/pull/544/files#diff-46fae3f9bc98017095df24cd6b014aa02cf1992249906d4d5ef92103857c11b4L46

Reply:
wow nice digging 😆 
Yeah I add the useEffect to clear the timeout 👍